--- Python-3.3.3/setup.py.orig	2013-11-21 18:36:59.259841950 +0100
+++ Python-3.3.3/setup.py	2013-11-21 19:07:23.173098741 +0100
@@ -835,7 +835,7 @@
         # a release.  Most open source OSes come with one or more
         # versions of BerkeleyDB already installed.
 
-        max_db_ver = (5, 3)
+        max_db_ver = (6, 2)
         min_db_ver = (3, 3)
         db_setup_debug = False   # verbose debug prints from this script?
 
--- Python-3.8.0/setup.py~	2019-10-15 10:03:10.000000000 +0200
+++ Python-3.8.0/setup.py	2019-10-15 10:03:52.493993277 +0200
@@ -1275,16 +1275,7 @@ class PyBuildExt(build_ext):
             for cand in dbm_order:
                 if cand == "ndbm":
                     if find_file("ndbm.h", self.inc_dirs, []) is not None:
-                        # Some systems have -lndbm, others have -lgdbm_compat,
-                        # others don't have either
-                        if self.compiler.find_library_file(self.lib_dirs,
-                                                               'ndbm'):
-                            ndbm_libs = ['ndbm']
-                        elif self.compiler.find_library_file(self.lib_dirs,
-                                                             'gdbm_compat'):
-                            ndbm_libs = ['gdbm_compat']
-                        else:
-                            ndbm_libs = []
+                        ndbm_libs = ['gdbm_compat']
                         if dbm_setup_debug: print("building dbm using ndbm")
                         dbmext = Extension('_dbm', ['_dbmmodule.c'],
                                            define_macros=[
