--- Python-3.3.0/Objects/obmalloc.c~	2012-09-29 10:00:46.000000000 +0200
+++ Python-3.3.0/Objects/obmalloc.c	2012-09-29 23:33:43.775178451 +0200
@@ -1846,7 +1846,9 @@
 
             if (p->ref.count == 0) {
                 /* currently unused */
+#ifdef Py_DEBUG
                 assert(pool_is_in_list(p, arenas[i].freepools));
+#endif
                 continue;
             }
             ++numpools[sz];
