--- Python-3.8.5/Makefile.pre.in.orig	2020-09-17 07:55:13.731893519 +0200
+++ Python-3.8.5/Makefile.pre.in	2020-09-17 10:33:36.683744945 +0200
@@ -224,6 +224,7 @@
 LIBRARY=	@LIBRARY@
 LDLIBRARY=      @LDLIBRARY@
 BLDLIBRARY=     @BLDLIBRARY@
+BLDLIBDIR=      @BLDLIBDIR@
 PY3LIBRARY=     @PY3LIBRARY@
 DLLLIBRARY=	@DLLLIBRARY@
 LDLIBRARYDIR=   @LDLIBRARYDIR@
@@ -588,7 +588,7 @@ clinic: check-clean-src $(srcdir)/Module
 
 # Build the interpreter
 $(BUILDPYTHON):	Programs/python.o $(LIBRARY_DEPS)
-	$(LINKCC) $(PY_CORE_LDFLAGS) $(LINKFORSHARED) -o $@ Programs/python.o $(BLDLIBRARY) $(LIBS) $(MODLIBS) $(SYSLIBS)
+	$(LINKCC) $(PY_CORE_LDFLAGS) $(LINKFORSHARED) -o $@ Programs/python.o $(BLDLIBDIR) $(BLDLIBRARY) $(LIBS) $(MODLIBS) $(SYSLIBS)
 
 platform: $(BUILDPYTHON) pybuilddir.txt
 	$(RUNSHARED) $(PYTHON_FOR_BUILD) -c 'import sys ; from sysconfig import get_platform ; print("%s-%d.%d" % (get_platform(), *sys.version_info[:2]))' >platform
@@ -723,7 +723,7 @@ Makefile Modules/config.c: Makefile.pre
 
 
 Programs/_testembed: Programs/_testembed.o $(LIBRARY_DEPS)
-	$(LINKCC) $(PY_CORE_LDFLAGS) $(LINKFORSHARED) -o $@ Programs/_testembed.o $(BLDLIBRARY) $(LIBS) $(MODLIBS) $(SYSLIBS)
+	$(LINKCC) $(PY_CORE_LDFLAGS) $(LINKFORSHARED) -o $@ Programs/_testembed.o $(BLDLIBDIR) $(BLDLIBRARY) $(LIBS) $(MODLIBS) $(SYSLIBS)
 
 ############################################################################
 # Importlib
--- Python-3.8.5/configure.ac.orig	2020-09-17 07:55:13.735226834 +0200
+++ Python-3.8.5/configure.ac	2020-09-17 10:46:44.172812084 +0200
@@ -1033,6 +1033,7 @@
 # with the ABI build flags appended.
 AC_SUBST(LDLIBRARY)
 AC_SUBST(DLLLIBRARY)
+AC_SUBST(BLDLIBDIR)
 AC_SUBST(BLDLIBRARY)
 AC_SUBST(PY3LIBRARY)
 AC_SUBST(LDLIBRARYDIR)
@@ -1040,6 +1041,7 @@
 AC_SUBST(RUNSHARED)
 AC_SUBST(LDVERSION)
 LDLIBRARY="$LIBRARY"
+BLDLIBDIR=''
 BLDLIBRARY='$(LDLIBRARY)'
 INSTSONAME='$(LDLIBRARY)'
 DLLLIBRARY=''
@@ -1156,7 +1158,8 @@
           ;;
     SunOS*)
 	  LDLIBRARY='libpython$(LDVERSION).so'
-	  BLDLIBRARY='-Wl,-R,$(LIBDIR) -L. -lpython$(LDVERSION)'
+	  BLDLIBDIR='-Wl,-R,$(LIBDIR) -L.'
+	  BLDLIBRARY='-lpython$(VERSION)'
 	  RUNSHARED=LD_LIBRARY_PATH=`pwd`${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
 	  INSTSONAME="$LDLIBRARY".$SOVERSION
 	  if test "$with_pydebug" != yes
@@ -1147,7 +1147,8 @@ if test $enable_shared = "yes"; then
           ;;
     Linux*|GNU*|NetBSD*|FreeBSD*|DragonFly*|OpenBSD*|VxWorks*)
 	  LDLIBRARY='libpython$(LDVERSION).so'
-	  BLDLIBRARY='-L. -lpython$(LDVERSION)'
+	  BLDLIBDIR='-Wl,-R,$(LIBDIR) -L.'
+	  BLDLIBRARY='-lpython$(VERSION)'
 	  RUNSHARED=LD_LIBRARY_PATH=`pwd`${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
 	  INSTSONAME="$LDLIBRARY".$SOVERSION
 	  if test "$with_pydebug" != yes
@@ -1183,12 +1187,14 @@
 			LDLIBRARY='libpython$(LDVERSION).sl'
 			;;
 	  esac
-	  BLDLIBRARY='-Wl,+b,$(LIBDIR) -L. -lpython$(LDVERSION)'
+	  BLDLIBDIR='-Wl,+b,$(LIBDIR) -L.'
+	  BLDLIBRARY='-lpython$(VERSION)'
 	  RUNSHARED=SHLIB_PATH=`pwd`${SHLIB_PATH:+:${SHLIB_PATH}}
 	  ;;
     Darwin*)
     	LDLIBRARY='libpython$(LDVERSION).dylib'
-	BLDLIBRARY='-L. -lpython$(LDVERSION)'
+	BLDLIBDIR='-L.'
+	BLDLIBRARY='-lpython$(LDVERSION)'
 	RUNSHARED=DYLD_LIBRARY_PATH=`pwd`${DYLD_LIBRARY_PATH:+:${DYLD_LIBRARY_PATH}}
 	;;
     AIX*)
