--- Python-3.3.1/Lib/platform.py~	2013-04-06 09:41:39.000000000 +0200
+++ Python-3.3.1/Lib/platform.py	2013-05-11 16:25:29.566791008 +0200
@@ -261,7 +261,7 @@
 _supported_dists = (
     'SuSE', 'debian', 'fedora', 'redhat', 'centos',
     'mandrake', 'mandriva', 'rocks', 'slackware', 'yellowdog', 'gentoo',
-    'UnitedLinux', 'turbolinux', 'arch', 'mageia')
+    'pld', 'UnitedLinux', 'turbolinux', 'arch', 'mageia')
 
 def _parse_release_file(firstline):
 
