--- Python-3.0.1/Modules/getpath.c.wiget	2009-02-25 10:58:09.000000000 +0100
+++ Python-3.0.1/Modules/getpath.c	2009-02-25 11:21:43.000000000 +0100
@@ -126,7 +126,8 @@
 
 #ifndef PYTHONPATH
 #define PYTHONPATH PREFIX "/" LIB_PYTHON VERSION ":" \
-              EXEC_PREFIX "/" LIB_PYTHON VERSION "/lib-dynload:"
+              EXEC_PREFIX "/" LIB_PYTHON VERSION "/lib-dynload:" \
+              PREFIX "/share/python" VERSION
 #endif
 
 #ifndef LANDMARK
@@ -138,6 +139,7 @@ static wchar_t exec_prefix[MAXPATHLEN+1]
 static wchar_t progpath[MAXPATHLEN+1];
 static wchar_t *module_search_path = NULL;
 static wchar_t lib_python[] = LIB_PYTHON VERSION;
+static wchar_t share_python[] = L"share/python" VERSION;
 
 /* In principle, this should use HAVE__WSTAT, and _wstat
    should be detected by autoconf. However, no current
@@ -338,7 +340,7 @@ search_for_prefix(wchar_t *argv0_path, w
         delim = wcschr(prefix, DELIM);
         if (delim)
             *delim = L'\0';
-        joinpath(prefix, lib_python);
+        joinpath(prefix, share_python);
         joinpath(prefix, LANDMARK);
         return 1;
     }
@@ -361,7 +363,7 @@ search_for_prefix(wchar_t *argv0_path, w
     copy_absolute(prefix, argv0_path);
     do {
         n = wcslen(prefix);
-        joinpath(prefix, lib_python);
+        joinpath(prefix, share_python);
         joinpath(prefix, LANDMARK);
         if (ismodule(prefix))
             return 1;
@@ -371,7 +373,7 @@ search_for_prefix(wchar_t *argv0_path, w
 
     /* Look at configure's PREFIX */
     wcsncpy(prefix, L"" PREFIX, MAXPATHLEN);
-    joinpath(prefix, lib_python);
+    joinpath(prefix, share_python);
     joinpath(prefix, LANDMARK);
     if (ismodule(prefix))
         return 1;
@@ -598,7 +600,7 @@ calculate_path(void)
             fprintf(stderr,
                 "Could not find platform independent libraries <prefix>\n");
         wcsncpy(prefix, L"" PREFIX, MAXPATHLEN);
-        joinpath(prefix, lib_python);
+        joinpath(prefix, share_python);
     }
     else
         reduce(prefix);
@@ -611,7 +613,7 @@ calculate_path(void)
     }
     else
         wcsncpy(zip_path, L"" PREFIX, MAXPATHLEN);
-    joinpath(zip_path, L"lib/python00.zip");
+    joinpath(zip_path, LIB_PYTHON "00.zip");
     bufsz = wcslen(zip_path);	/* Replace "00" with version */
     zip_path[bufsz - 6] = VERSION[0];
     zip_path[bufsz - 5] = VERSION[2];
--- Python-3.0.1/Lib/site.py.wiget	2009-02-25 10:58:09.000000000 +0100
+++ Python-3.0.1/Lib/site.py	2009-02-25 11:03:48.000000000 +0100
@@ -262,11 +262,10 @@ def addsitepackages(known_paths):
                                         "python" + sys.version[:3],
                                         "site-packages"))
             sitedirs.append(os.path.join(prefix, sys.lib, "site-python"))
-            if sys.lib != 'lib':
-                sitedirs.append(os.path.join(prefix, 'lib',
-                                            "python" + sys.version[:3],
-                                            "site-packages"))
-                sitedirs.append(os.path.join(prefix, 'lib', "site-python"))
+            sitedirs.append(os.path.join(prefix, 'share',
+                                        "python" + sys.version[:3],
+                                        "site-packages"))
+            sitedirs.append(os.path.join(prefix, 'share', "site-python"))
         	
         else:
             sitedirs.append(prefix)
--- Python-3.0.1/Lib/distutils/sysconfig.py.wiget	2009-02-25 10:58:09.000000000 +0100
+++ Python-3.0.1/Lib/distutils/sysconfig.py	2009-02-25 10:58:09.000000000 +0100
@@ -114,12 +114,12 @@ def get_python_lib(plat_specific=0, stan
         prefix = plat_specific and EXEC_PREFIX or PREFIX
 
     if os.name == "posix":
-        if plat_specific:
-            lib = sys.lib
+        if plat_specific: 
+            libpython = os.path.join(prefix,
+                                 sys.lib, "python" + get_python_version())
         else:
-            lib = 'lib'
-        libpython = os.path.join(prefix,
-                                 lib, "python" + get_python_version())
+            libpython = os.path.join(prefix,
+                                 "share", "python" + get_python_version())
         if standard_lib:
             return libpython
         else:
--- Python-3.0.1/Lib/distutils/command/install.py.wiget	2009-02-25 10:58:09.000000000 +0100
+++ Python-3.0.1/Lib/distutils/command/install.py	2009-02-25 10:58:09.000000000 +0100
@@ -40,14 +40,14 @@ else:
 
 INSTALL_SCHEMES = {
     'unix_prefix': {
-        'purelib': '$base/lib/python$py_version_short/site-packages',
+        'purelib': '$base/share/python$py_version_short/site-packages',
         'platlib': '$platbase/'+libname+'/python$py_version_short/site-packages',
         'headers': '$base/include/python$py_version_short/$dist_name',
         'scripts': '$base/bin',
         'data'   : '$base',
         },
     'unix_home': {
-        'purelib': '$base/lib/python',
+        'purelib': '$base/share/python',
         'platlib': '$base/'+libname+'/python',
         'headers': '$base/include/python/$dist_name',
         'scripts': '$base/bin',
--- Python-3.0.1/Makefile.pre.in.wiget	2009-02-25 10:58:09.000000000 +0100
+++ Python-3.0.1/Makefile.pre.in	2009-02-25 10:58:09.000000000 +0100
@@ -93,7 +93,7 @@ LIBDIR=		$(exec_prefix)/$(LIB)
 MANDIR=		@mandir@
 INCLUDEDIR=	@includedir@
 CONFINCLUDEDIR=	$(exec_prefix)/include
-SCRIPTDIR=	$(prefix)/$(LIB)
+SCRIPTDIR=	$(prefix)/share
 
 # Detailed destination directories
 BINLIBDEST=	$(LIBDIR)/python$(VERSION)
--- Python-3.0.1/Lib/distutils/tests/test_install.py.wig	2009-02-25 11:42:38.000000000 +0100
+++ Python-3.0.1/Lib/distutils/tests/test_install.py	2009-02-25 11:51:23.000000000 +0100
@@ -38,7 +38,7 @@ class InstallTestCase(support.TempdirMan
             expected = os.path.normpath(expected)
             self.assertEqual(got, expected)
 
-        libdir = os.path.join(destination, "lib", "python")
+        libdir = os.path.join(destination, "share", "python")
         platlibdir =  os.path.join(destination, sys.lib, "python")
         check_path(cmd.install_lib, libdir)
         check_path(cmd.install_platlib, platlibdir)
