--- Python-3.0.1/Lib/cgi.py.wiget	2008-12-04 21:11:57.000000000 +0100
+++ Python-3.0.1/Lib/cgi.py	2009-03-14 18:12:32.000000000 +0100
@@ -1,6 +1,6 @@
-#! /usr/local/bin/python
+#! /usr/bin/python3
 
-# NOTE: the above "/usr/local/bin/python" is NOT a mistake.  It is
+# NOTE: the above "/usr/bin/python3" is NOT a mistake.  It is
 # intentionally NOT "/usr/bin/env python".  On many systems
 # (e.g. Solaris), /usr/local/bin is not in $PATH as passed to CGI
 # scripts, and /usr/local/bin is the default directory where Python is

