--- Python-3.5.0/configure.ac~	2015-09-13 13:41:26.000000000 +0200
+++ Python-3.5.0/configure.ac	2015-09-13 20:06:11.597280920 +0200
@@ -1258,16 +1258,16 @@ then
 		# Optimization messes up debuggers, so turn it off for
 		# debug builds.
                 if "$CC" -v --help 2>/dev/null |grep -- -Og > /dev/null; then
-                    OPT="-g -Og -Wall $STRICT_PROTO"
+                    OPT="-g $CPPFLAGS $CFLAGS -Wall $STRICT_PROTO"
                 else
-                    OPT="-g -O0 -Wall $STRICT_PROTO"
+                    OPT="-g $CPPFLAGS $CFLAGS -Wall $STRICT_PROTO"
                 fi
 	    else
-		OPT="-g $WRAP -O3 -Wall $STRICT_PROTO"
+		OPT="-g $WRAP $CPPFLAGS $CFLAGS -Wall $STRICT_PROTO"
 	    fi
 	    ;;
 	*)
-	    OPT="-O3 -Wall $STRICT_PROTO"
+	    OPT="$CPPFLAGS $CFLAGS -Wall $STRICT_PROTO"
 	    ;;
 	esac
 	case $ac_sys_system in
