--- ./Tools/scripts/run_tests.py~	2018-06-27 05:07:35.000000000 +0200
+++ ./Tools/scripts/run_tests.py	2018-06-29 09:40:05.838625461 +0200
@@ -31,7 +31,6 @@ def main(regrtest_args):
     args.extend(test.support.args_from_interpreter_flags())
 
     args.extend(['-m', 'test',    # Run the test suite
-                 '-r',            # Randomize test order
                  '-w',            # Re-run failed tests in verbose mode
                  ])
     if sys.platform == 'win32':
