--- a/Makefile.pre.in~	2024-01-12 14:35:45.000000000 +0100
+++ b/Makefile.pre.in	2024-01-12 14:40:01.203324556 +0100
@@ -1921,7 +1921,7 @@ altinstall: commoninstall
 .PHONY: commoninstall
 commoninstall:  check-clean-src @FRAMEWORKALTINSTALLFIRST@ \
 		altbininstall libinstall inclinstall libainstall \
-		sharedinstall altmaninstall @FRAMEWORKALTINSTALLLAST@
+		sharedinstall altmaninstall installcompile @FRAMEWORKALTINSTALLLAST@
 
 # Install shared libraries enabled by Setup
 DESTDIRS=	$(exec_prefix) $(LIBDIR) $(BINLIBDEST) $(DESTSHARED)
@@ -2576,6 +2576,8 @@ libinstall:	all $(srcdir)/Modules/xxmodu
 		echo "Applying app store compliance patch"; \
 		patch --force --reject-file "$(abs_builddir)/app-store-compliance.rej" --strip 2 --directory "$(DESTDIR)$(LIBDEST)" --input "$(abs_srcdir)/$(APP_STORE_COMPLIANCE_PATCH)" || true ; \
 	fi
+.PHONY: installcompile
+installcompile: libainstall libinstall
 	@ # Build PYC files for the 3 optimization levels (0, 1, 2)
 	-PYTHONPATH=$(DESTDIR)$(LIBDEST) $(RUNSHARED) \
 		$(PYTHON_FOR_BUILD) -Wi $(DESTDIR)$(LIBDEST)/compileall.py \

