--- Python-3.7.4/Makefile.pre.in~	2019-08-05 14:10:46.000000000 +0200
+++ Python-3.7.4/Makefile.pre.in	2019-08-05 14:37:01.691326225 +0200
@@ -260,7 +260,7 @@ TCLTK_LIBS=	@TCLTK_LIBS@
 # The task to run while instrumented when building the profile-opt target.
 # We exclude unittests with -x that take a rediculious amount of time to
 # run in the instrumented training build or do not provide much value.
-PROFILE_TASK=-m test.regrtest --pgo
+PROFILE_TASK=-m test.regrtest --pgo $(TESTOPTS)
 
 # report files for gcov / lcov coverage report
 COVERAGE_INFO=	$(abs_builddir)/coverage.info
