--- Python-3.3.3/setup.py.orig	2013-11-21 18:36:59.259841950 +0100
+++ Python-3.3.3/setup.py	2013-11-21 19:07:23.173098741 +0100
@@ -835,7 +835,7 @@
         # a release.  Most open source OSes come with one or more
         # versions of BerkeleyDB already installed.
 
-        max_db_ver = (5, 3)
+        max_db_ver = (6, 1)
         min_db_ver = (3, 3)
         db_setup_debug = False   # verbose debug prints from this script?
 
