--- Python-3.6.2/configure.ac~	2017-07-08 05:33:27.000000000 +0200
+++ Python-3.6.2/configure.ac	2017-08-07 10:28:44.498951134 +0200
@@ -1491,24 +1491,7 @@ then
             CFLAGS_ALIASING="-fno-strict-aliasing"
         fi
 
-	case $ac_cv_prog_cc_g in
-	yes)
-	    if test "$Py_DEBUG" = 'true' ; then
-		# Optimization messes up debuggers, so turn it off for
-		# debug builds.
-                if "$CC" -v --help 2>/dev/null |grep -- -Og > /dev/null; then
-                    OPT="-g -Og -Wall"
-                else
-                    OPT="-g -O0 -Wall"
-                fi
-	    else
-		OPT="-g $WRAP -O3 -Wall"
-	    fi
-	    ;;
-	*)
-	    OPT="-O3 -Wall"
-	    ;;
-	esac
+	OPT="$OPT $CPPFLAGS $CFLAGS"
 
 	case $ac_sys_system in
 	    SCO_SV*) OPT="$OPT -m486 -DSCO5"

