diff -urNp -x '*.orig' Python-3.9.2.org/Lib/distutils/sysconfig.py Python-3.9.2/Lib/distutils/sysconfig.py
--- Python-3.9.2.org/Lib/distutils/sysconfig.py	2021-02-25 18:40:49.900518043 +0100
+++ Python-3.9.2/Lib/distutils/sysconfig.py	2021-02-25 18:40:51.570570064 +0100
@@ -260,7 +260,7 @@ def get_makefile_filename():
     """Return full pathname of installed Makefile from the Python build."""
     if python_build:
         return os.path.join(_sys_home or project_base, "Makefile")
-    lib_dir = get_python_lib(plat_specific=0, standard_lib=1)
+    lib_dir = get_python_lib(plat_specific=1, standard_lib=1)
     config_file = 'config-{}{}'.format(get_python_version(), build_flags)
     if hasattr(sys.implementation, '_multiarch'):
         config_file += '-%s' % sys.implementation._multiarch
