diff -urNp -x '*.orig' Python-3.9.2.org/Lib/test/test_compileall.py Python-3.9.2/Lib/test/test_compileall.py
--- Python-3.9.2.org/Lib/test/test_compileall.py	2021-02-19 13:31:44.000000000 +0100
+++ Python-3.9.2/Lib/test/test_compileall.py	2021-02-25 18:40:29.876560976 +0100
@@ -419,6 +419,7 @@ class EncodingTest(unittest.TestCase):
             sys.stdout = orig_stdout
 
 
+@unittest.skipUnless(support.is_resource_enabled("cmdline"), "python cmdline tests disabled")
 class CommandLineTestsBase:
     """Test compileall's CLI."""
 
diff -urNp -x '*.orig' Python-3.9.2.org/Lib/test/test_warnings/__init__.py Python-3.9.2/Lib/test/test_warnings/__init__.py
--- Python-3.9.2.org/Lib/test/test_warnings/__init__.py	2021-02-19 13:31:44.000000000 +0100
+++ Python-3.9.2/Lib/test/test_warnings/__init__.py	2021-02-25 18:40:29.876560976 +0100
@@ -1126,6 +1126,7 @@ class PyCatchWarningTests(CatchWarningTe
     module = py_warnings
 
 
+@unittest.skipUnless(support.is_resource_enabled("cmdline"), "python cmdline tests disabled")
 class EnvironmentVariableTests(BaseTest):
 
     def test_single_warning(self):
@@ -1278,9 +1278,11 @@ class EnvironmentVariableTests(BaseTest)
             PYTHONDEVMODE="")
         self.assertEqual(stdout, str([PYTHONWARNINGS]).encode())
 
+@unittest.skipUnless(support.is_resource_enabled("cmdline"), "python cmdline tests disabled")
 class CEnvironmentVariableTests(EnvironmentVariableTests, unittest.TestCase):
     module = c_warnings
 
+@unittest.skipUnless(support.is_resource_enabled("cmdline"), "python cmdline tests disabled")
 class PyEnvironmentVariableTests(EnvironmentVariableTests, unittest.TestCase):
     module = py_warnings
 
@@ -1326,6 +1328,7 @@ class _DeprecatedTest(BaseTest, unittest
 
 class BootstrapTest(unittest.TestCase):
 
+    @unittest.skipUnless(support.is_resource_enabled("cmdline"), "python cmdline tests disabled")
     def test_issue_8766(self):
         # "import encodings" emits a warning whereas the warnings is not loaded
         # or not completely loaded (warnings imports indirectly encodings by
