--- Python-3.8.0/Makefile.pre.in~	2019-10-15 10:04:12.000000000 +0200
+++ Python-3.8.0/Makefile.pre.in	2019-10-15 10:09:54.901153005 +0200
@@ -260,7 +260,7 @@ TCLTK_LIBS=	@TCLTK_LIBS@
 # To speed up profile generation, we don't run the full unit test suite
 # by default. The default is "-m test --pgo". To run more tests, use
 # PROFILE_TASK="-m test --pgo-extended"
-PROFILE_TASK=	@PROFILE_TASK@
+PROFILE_TASK=	@PROFILE_TASK@ $(TESTOPTS)
 
 # report files for gcov / lcov coverage report
 COVERAGE_INFO=	$(abs_builddir)/coverage.info
