--- Python-3.0.1/Lib/distutils/tests/test_build_scripts.py.wiget	2004-08-26 07:44:02.000000000 +0200
+++ Python-3.0.1/Lib/distutils/tests/test_build_scripts.py	2009-03-14 18:12:32.000000000 +0100
@@ -58,7 +58,7 @@ class BuildScriptsTestCase(support.Tempd
                            "pass\n"))
         expected.append("script2.py")
         self.write_script(dir, "script2.py",
-                          ("#!/usr/bin/python\n"
+                          ("#!/usr/bin/python3\n"
                            "# bogus script w/ Python sh-bang\n"
                            "pass\n"))
         expected.append("shell.sh")
--- Python-3.0.1/Lib/distutils/tests/test_install_scripts.py.wiget	2004-08-03 20:53:07.000000000 +0200
+++ Python-3.0.1/Lib/distutils/tests/test_install_scripts.py	2009-03-14 18:12:32.000000000 +0100
@@ -48,7 +48,7 @@ class InstallScriptsTestCase(support.Tem
         write_script("script1.py", ("#! /usr/bin/env python2.3\n"
                                     "# bogus script w/ Python sh-bang\n"
                                     "pass\n"))
-        write_script("script2.py", ("#!/usr/bin/python\n"
+        write_script("script2.py", ("#!/usr/bin/python3\n"
                                     "# bogus script w/ Python sh-bang\n"
                                     "pass\n"))
         write_script("shell.sh", ("#!/bin/sh\n"
--- Python-3.0.1/Lib/cgi.py.wiget	2008-12-04 21:11:57.000000000 +0100
+++ Python-3.0.1/Lib/cgi.py	2009-03-14 18:12:32.000000000 +0100
@@ -1,6 +1,6 @@
-#! /usr/local/bin/python
+#! /usr/bin/python3
 
-# NOTE: the above "/usr/local/bin/python" is NOT a mistake.  It is
+# NOTE: the above "/usr/bin/python3" is NOT a mistake.  It is
 # intentionally NOT "/usr/bin/env python".  On many systems
 # (e.g. Solaris), /usr/local/bin is not in $PATH as passed to CGI
 # scripts, and /usr/local/bin is the default directory where Python is

