--- Python-3.7.4/Misc/python-config.sh.in.orig	2019-07-08 20:03:50.000000000 +0200
+++ Python-3.7.4/Misc/python-config.sh.in	2019-08-03 07:45:05.226390619 +0200
@@ -92,7 +92,7 @@
             if [ "$PY_ENABLE_SHARED" = "0" ] ; then
                 LIBPLUSED="-L$LIBPL"
             fi
-            echo "$LIBPLUSED -L$libdir $LIBS"
+            echo "$LIBPLUSED $LIBS"
         ;;
         --extension-suffix)
             echo "$SO"
