--- Python-3.1/Modules/python.c~	2009-05-29 18:22:26.000000000 +0200
+++ Python-3.1/Modules/python.c	2009-06-29 16:18:12.800560478 +0200
@@ -57,7 +57,7 @@
 	/* Overallocate; as multi-byte characters are in the argument, the
 	   actual output could use less memory. */
 	argsize = strlen(arg) + 1;
-	res = PyMem_Malloc(argsize*sizeof(wchar_t));
+	res = (wchar_t *)PyMem_Malloc(argsize*sizeof(wchar_t));
 	if (!res) goto oom;
 	in = (unsigned char*)arg;
 	out = res;
